homeassistant/components/switch/insteon_plm.py

```diff
@@ -63,6 +63,12 @@ def async_turn_off(self, **kwargs):
 class InsteonPLMOpenClosedDevice(InsteonPLMEntity, SwitchDevice):
     """A Class for an Insteon device."""
 
+    @property
+    def is_on(self):
+        """Return the boolean response if the node is on."""
+        onlevel = self._insteon_device_state.value
```

Comment:
Is this doing I/O ?

Reply:
No. The underlying library captures the state change asynchronously. It savesthe state of the device in the device state value variable so this represents the persistent value of the state.